xamarin/Xamarin.Forms
# [Android] TalkBack now reads name and helptext on buttons.

### Description of Change ###

Original PR here
https://github.com/xamarin/Xamarin.Forms/pull/9728

- Removed fix on original PR for fixing AutomationId and Content Description. That fix needs more review
- This PR applies a different fix to resolve the wrong name/help text being read. The main problem is that the Renderers for Switch/Button override the SetBasicContentDescription from this PR https://github.com/xamarin/Xamarin.Forms/pull/4094. This was done so that the Hint doesn't get set on the Switch/Button but it needs to use the actual control and not the ViewGroup container
- Fixed an issue where a reused control would get set to the previous elements AutomationId. The defaultContentDescription would only get set when ContentDescription wasn't null. This leads to the defaultContentDescription getting set to the AutomationId because you can't just use a null check to determine if a default should be set for a string. This would make it so when a control gets recycled to a new element it would set the ContentDescription to the previous AutomationId thinking that was the `defaultContentDescription`. The defaultContentDescription/defaultHint are now set only once explicitly the first time the AutomationId/ContentDescription are set. 

### Issues Resolved ### 
- fixes #5150

### Platforms Affected ### 
- Android


### Testing Procedure ###
Switch to Legacy Renderers as the change is only relevant for the Legacy Button Renderer. Switch Renderer doesn't have a fast renderer so it's relevant either way

- Verify UI test 5150
- Verify UI Test 4053 

### PR Checklist ###
